server/settings_test.go

```diff
@@ -72,6 +77,193 @@ func TestSettings(t *testing.T) {
+
+				res, err := serverClient.Default.Server.ChangeSettings(&server.ChangeSettingsParams{
+					Body: server.ChangeSettingsBody{
+						DisableStt:      true,
+						EnableTelemetry: true,
+					},
+					Context: pmmapitests.Context,
+				})
+				require.NoError(t, err)
+				assert.False(t, res.Payload.Settings.SttEnabled)
+				assert.True(t, res.Payload.Settings.TelemetryEnabled)
+				assert.Empty(t, err)
+
+				resg, err := serverClient.Default.Server.GetSettings(nil)
+				require.NoError(t, err)
+				assert.True(t, resg.Payload.Settings.TelemetryEnabled)
+				assert.False(t, resg.Payload.Settings.SttEnabled)
+			})
+
+			t.Run("DisableSTTAndDisableTelemetry", func(t *testing.T) {
+				defer teardown(t)
+
+				res, err := serverClient.Default.Server.ChangeSettings(&server.ChangeSettingsParams{
+					Body: server.ChangeSettingsBody{
+						DisableStt:       true,
+						DisableTelemetry: true,
+					},
+					Context: pmmapitests.Context,
+				})
+				require.NoError(t, err)
+				assert.False(t, res.Payload.Settings.SttEnabled)
+				assert.False(t, res.Payload.Settings.TelemetryEnabled)
+				assert.Empty(t, err)
+
+				resg, err := serverClient.Default.Server.GetSettings(nil)
+				require.NoError(t, err)
+				assert.False(t, resg.Payload.Settings.TelemetryEnabled)
+				assert.False(t, resg.Payload.Settings.SttEnabled)
+			})
+
+			t.Run("EnableSTTWhileTelemetryEnabled", func(t *testing.T) {
+				defer teardown(t)
+
+				// Ensure Telemetry is enabled
+				res, err := serverClient.Default.Server.ChangeSettings(&server.ChangeSettingsParams{
+					Body: server.ChangeSettingsBody{
+						EnableTelemetry: true,
+					},
+					Context: pmmapitests.Context,
+				})
+				require.NoError(t, err)
+				assert.True(t, res.Payload.Settings.TelemetryEnabled)
+
+				res, err = serverClient.Default.Server.ChangeSettings(&server.ChangeSettingsParams{
+					Body: server.ChangeSettingsBody{
+						EnableStt: true,
+					},
+					Context: pmmapitests.Context,
+				})
+				require.NoError(t, err)
+
+				assert.True(t, res.Payload.Settings.SttEnabled)
+				assert.Empty(t, err)
+
+				resg, err := serverClient.Default.Server.GetSettings(nil)
+				require.NoError(t, err)
+				assert.True(t, resg.Payload.Settings.TelemetryEnabled)
+				assert.True(t, resg.Payload.Settings.SttEnabled)
+			})
+
+			t.Run("DisableSTTWhileItIsDisabled", func(t *testing.T) {
+				defer teardown(t)
+
+				res, err := serverClient.Default.Server.ChangeSettings(&server.ChangeSettingsParams{
+					Body: server.ChangeSettingsBody{
+						DisableStt: true,
+					},
+					Context: pmmapitests.Context,
+				})
+				require.NoError(t, err)
+				assert.False(t, res.Payload.Settings.SttEnabled)
+				assert.Empty(t, err)
+
+				resg, err := serverClient.Default.Server.GetSettings(nil)
+				require.NoError(t, err)
+				assert.True(t, resg.Payload.Settings.TelemetryEnabled)
+				assert.False(t, resg.Payload.Settings.SttEnabled)
+			})
+			t.Run("STTEnabledState", func(t *testing.T) {
```

Comment:
Let's make the same kind of tests with disabled telemetry.

Reply:
Done